lib/fs/filec.mli

```diff
@@ -0,0 +1,20 @@
+(** File contents as an array of lines, where each line is wrapped into a
+    document (for rendering efficiency) *)
+type t =
+  | Binary
+  | Text of {
+      lines : Pretty.doc array;
+      offset : int;
+    }
+
+(** Reads file contents using 'bat' to have pretty syntax highlighting **)
+val read_file_contents : string -> t
+
+(** Returns offset based on file type of contents **)
+val offset_from_file_contents : t -> int
+
+(** Returns the len of file contents **)
+val line_len_from_file_contents : t -> int
+
+(** Returns the lines of file contents **)
+val lines_from_file_contents : t -> Pretty.doc array
```

Comment:
Let's simplify names of the functions here. Since they're prefixed with `Filec`, there's no need to add the `file_contents` suffix.

This is a typical trick in OCaml to use modules as kinda namespaces. So these functions will become:

```ocaml
Filec.read
Filec.offset
Filec.length
Filec.lines
```